src/main/scala/ai/privado/utility/SQLParser.scala

```diff
@@ -137,29 +133,29 @@ object SQLParser {
     }
   }
 
-  def getColumns(plainSelect: PlainSelect): List[SQLColumn] = {
+  def getColumns(plainSelect: PlainSelect, sqlTable: SQLTable): List[SQLColumn] = {
     plainSelect.getSelectItems.asScala.flatMap { (item: SelectItem) =>
       item.toString match {
         case f: String if f.contains("(") =>
           val function = CCJSqlParserUtil.parseExpression(f).asInstanceOf[Function]
-          function.getParameters.getExpressions.asScala.map(column => createSQLColumnItem(column))
+          function.getParameters.getExpressions.asScala.map(column => createSQLColumnItem(column, sqlTable))
         case _ =>
-          List(createSQLColumnItem(item))
+          List(createSQLColumnItem(item, sqlTable))
       }
     }.toList
   }
 
-  private def createSQLTableItem(table: Table) = {
+  private def createSQLTableItem(table: Table): SQLTable = {
     val tableName: String = table.getName
     val tableLineNumber   = Try(table.getASTNode.jjtGetFirstToken().beginLine).getOrElse(NUMBER_ONE)
     val tableColumnNumber = Try(table.getASTNode.jjtGetFirstToken().beginColumn).getOrElse(NUMBER_MINUSONE)
     SQLTable(tableName, tableLineNumber, tableColumnNumber)
   }
 
-  private def createSQLColumnItem(column: ASTNodeAccess) = {
+  private def createSQLColumnItem(column: ASTNodeAccess, sqlTable: SQLTable) = {
     SQLColumn(
       column.toString,
-      Try(column.getASTNode.jjtGetFirstToken().beginLine).getOrElse(NUMBER_ONE),
+      Try(column.getASTNode.jjtGetFirstToken().beginLine).getOrElse(NUMBER_ONE) + sqlTable.lineNumber,
```

Comment:
Is this yielding correct result, shouldn't this be `sqlTable.lineNumber - 1` ?

Reply:
you are correct missed adding `-1`
addressed here: feed56c0872054407657c8cb8bef4a2daf9d98cd